src/newgrf.cpp

```diff
@@ -7818,8 +7813,7 @@ static bool HandleParameterInfo(ByteReader *buf)
 
 		if (id >= _cur.grfconfig->param_info.size()) {
 			uint num_to_add = id - _cur.grfconfig->param_info.size() + 1;
-			GRFParameterInfo **newdata = _cur.grfconfig->param_info.Append(num_to_add);
-			MemSetT<GRFParameterInfo *>(newdata, 0, num_to_add);
+			while (num_to_add--) _cur.grfconfig->param_info.push_back(nullptr);
```

Comment:
yesh, that was an ugly thing. Not sure how much better this one is though...
Also, NULL for now - nullptr comes next :)